devito/types/tensor.py

```diff
@@ -169,7 +182,12 @@ def root_dimensions(self):
     @cached_property
     def space_order(self):
         """The space order for all components."""
-        return ({a.space_order for a in self} - {None}).pop()
+        orders = self.applyfunc(lambda x: x.space_order)
+        if len(set(orders)) > 1:
+            return orders
```

Comment:
does this even make sense? shouldn't the space_order be homogeneous ?

Reply:
In most cases yes, but I don't think it's mandatory no. There is nowhere in the codebase where we actually require it since everything is processed by single component.

Reply:
sure, but would you not expect, from a mathematical point of view, the components to be homogeneous or "compatible"? basically the same restriction I have in Bundle

Reply:
You actually might want not too. Especially when you think about things with different grid spacings in x/y/z where you might want to have different orders too for the vertical/horizontal components